### PR TITLE
Improvements for of email-templates

### PR DESCRIPTION
**Pull request by janwo**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: https://github.com/crocodilejs/node-email-templates/blob/master/examples/handlebars_i18n/index.js
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Comment by janwo** [comment]
Retry, see comments of https://github.com/DefinitelyTyped/DefinitelyTyped/pull/18166m

**Review by cyrilschumacher — APPROVED** [review]

**Comment by RyanCavanaugh** [comment]
Welcome back Travis!

**Comment by janwo** [comment]
@RyanCavanaugh Done! 👍

**Review by cyrilschumacher — APPROVED** [review]

File: types/email-templates/index.d.ts [hunk 1]
```diff
@@ -62,8 +62,29 @@ declare module "email-templates" {
         /**
          * @summary Render a single template.
          * @param {EmailTemplateCallback|Object} locals The variables or callback function.
+         */
+        render(locals: EmailTemplateCallback|Object): Promise<EmailTemplateResults>;
```
Comment (gurisko) [inline types/email-templates/index.d.ts]: This doesn't return a promise if `EmailTemplateCallback` is passed.

File: types/email-templates/index.d.ts [hunk 2]
```diff
@@ -62,8 +62,29 @@ declare module "email-templates" {
...
+         * @param {EmailTemplateCallback} callback The callback function.
+         */
+        render(locals: EmailTemplateCallback|Object, callback: EmailTemplateCallback): void;
+
+        /**
+         * @summary Render a single template.
+         * @param {EmailTemplateCallback|Object} locals The variables or callback function.
+         * @param {EmailTemplateCallback} locale The locale string.
+         */
+        render(locals: EmailTemplateCallback|Object, locale: string): Promise<EmailTemplateResults>;
```
Comment (gurisko) [inline types/email-templates/index.d.ts]: `EmailTemplateCallback` shouldn't be here.

File: types/email-templates/index.d.ts [hunk 3]
```diff
@@ -62,8 +62,29 @@ declare module "email-templates" {
...
          * @param {EmailTemplateCallback|Object} locals The variables or callback function.
+         */
+        render(locals: EmailTemplateCallback|Object): Promise<EmailTemplateResults>;
+
+        /**
+         * @summary Render a single template.
+         * @param {EmailTemplateCallback|Object} locals The variables or callback function.
+         * @param {EmailTemplateCallback} callback The callback function.
+         */
+        render(locals: EmailTemplateCallback|Object, callback: EmailTemplateCallback): void;
```
Comment (gurisko) [inline types/email-templates/index.d.ts]: This accepts two callback functions.

**Review by gurisko — CHANGES_REQUESTED** [review]

**Comment by gurisko** [comment]
These changes do not comply with the latest release 2.6.0. Please have a look at #18254

**Pull request closed without merging**